astropy/utils/masked/core.py

```diff
@@ -1078,10 +1080,18 @@ def argsort(self, axis=-1, kind=None, order=None):
 
         return np.lexsort(keys, axis=axis)
 
-    def sort(self, axis=-1, kind=None, order=None):
-        """Sort an array in-place. Refer to `numpy.sort` for full documentation."""
+    def sort(self, axis=-1, kind=None, order=None, *, stable=False):
+        """Sort an array in-place. Refer to `numpy.sort` for full documentation.
+
+        Notes
+        -----
+        stable is only used with numpy 2.0 and newer, and is ignored otherwise.
+        """
         # TODO: probably possible to do this faster than going through argsort!
-        indices = self.argsort(axis, kind=kind, order=order)
+        argsort_kwargs = dict(kind=kind, order=order)
+        if not NUMPY_LT_2_0:
```

Comment:
By setting the default to `np._NoValue`, we can change this to
```
if stable is not np._NoValue:
   argsort_kwargs["stable"] = stable
```
I do like that you pass them on, since in principle a subclass could override `argsort` and do something with it.